department-of-veterans-affairs/caseflow
# Enable users with access to correct issues on a substitute appeal

Resolves [jira ticket 1764](https://vajira.max.gov/browse/CASEFLOW-1764), which states that BvaIntake and COB users should be able to edit substitute appeal's issues through the "correct issues" link.  Before this PR, a bug made that impossible.

### Description

During a slack call I had with Ann-Marie and Paula on Wednesday, July 7, Ann-Marie demonstrated that intake users select request issues' decision dates during the intake process.  She also stated that when request issues are copied over to a substitute appeal, the request issues should retain their original decision date.

Previously, request issues' decision dates were set to equal "nil" during the substitute appellant process. Previous code [here](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/appellant_substitution.rb#L77). I discussed this with Yoom on a call Monday, July 17, and he said he thinks that line 77 was a bug.  He did suggest reaching out to Foxtrot about this, as they have the most domain knowledge around request issues.  I asked in slack for Foxtrot to confirm my [understanding of request issue decision dates](https://dsva.slack.com/archives/C54QCEHAL/p1626726976108200).

Also, I want to share [a previous helpful slack conversation with Yang](https://dsva.slack.com/archives/CJL810329/p1625680764377900) about request issue decision dates.  He stated that if it is a non-rating request issue, there should always be a decision date.  There will sometimes be request issues in the wild that don't have decision dates, which is tech debt for Foxtrot.

### Acceptance Criteria
- [ ] A COB and BvaIntake user can click "correct issues" on a substitute appeal and successfully edit request issues

### Testing Plan
1. Switch to being a Clerk of the Board user that also belongs to the BvaIntake group.  If you can't find any such users, switch to a clerk of the board user, and then add the user to the BvaIntake group.
````
bvatcobb = User.find_by(css_id: "BVATCOBB")
BvaIntake.singleton.add_user(bvatcobb)
````
2. Create an appeal that has both request and decision issues.  Mark the veteran as deceased.
````
appeal = FactoryBot.create(:appeal, :dispatched, :with_request_issues, :with_decision_issue, disposition: "dismissed_death")
veteran = appeal.veteran
veteran.update(date_of_death: Time.now)
````
3. Go through the appellant substitution flow and click on "Correct issues" on the substitute appeal's case details page.
4. Confirm that you can click on "Correct issues" without seeing any error messages.
5. Confirm that you can delete an issue.

Here is a demo of this work in action.

https://user-images.githubusercontent.com/80347702/126228131-5414bac2-c2cc-4e4a-9a06-1a160d30d61d.mov
